backend/src/LeanCode.AppRating/Processes/SendEmailOnLowRateSubmitted.cs

```diff
@@ -0,0 +1,46 @@
+using LeanCode.AppRating.Configuration;
+using LeanCode.AppRating.EmailViewModels;
+using LeanCode.SendGrid;
+using MassTransit;
+using SendGrid.Helpers.Mail;
+using Serilog;
+
+namespace LeanCode.AppRating.Processes;
+
+public class SendEmailOnLowRateSubmitted<TUserId> : IConsumer<RateSubmitted<TUserId>>
+    where TUserId : notnull, IEquatable<TUserId>
+{
+    private readonly ILogger logger = Log.ForContext<SendEmailOnLowRateSubmitted<TUserId>>();
+    private readonly SendGridRazorClient sendGridRazorClient;
+    private readonly AdminConfiguration adminConfiguration;
+
+    public SendEmailOnLowRateSubmitted(SendGridRazorClient sendGridRazorClient, AdminConfiguration adminConfiguration)
+    {
+        this.sendGridRazorClient = sendGridRazorClient;
+        this.adminConfiguration = adminConfiguration;
+    }
+
+    public async Task Consume(ConsumeContext<RateSubmitted<TUserId>> context)
+    {
+        if (context.Message.Rating > adminConfiguration.MaximalLowRatingInclusive)
```

Comment:
Let's move this check to the command handler? I.e. publish `LowRating` event when the rating is, well, low?

Reply:
I had in mind that we may need info about all ratings in the future, and the consumer should decide if the message matches it's criteria. 
I can change it the way you describe above, but I can't see how this is better